pkg/did/creator/creator.go

```diff
@@ -4,79 +4,104 @@ Copyright Avast Software. All Rights Reserved.
-	// DIDMethodKey is the name used by the Create method for the did:key method.
+	// DIDMethodKey is the name recognized by the Create method for the did:key method.
 	DIDMethodKey = "key"
-
-	didMethodPrefix            = "did:"
-	ed25519VerificationKey2018 = "Ed25519VerificationKey2018"
+	// Ed25519VerificationKey2018 is a supported DID verification type.
+	Ed25519VerificationKey2018 = "Ed25519VerificationKey2018"
 )
 
-// DIDCreator is used for creating DID Documents using supported DID methods.
-type DIDCreator struct {
-	keyHandleReader api.KeyHandleReader
+// Creator is used for creating DID Documents using supported DID methods.
+type Creator struct {
+	keyWriter api.KeyWriter
+	keyReader api.KeyReader
+}
+
+// NewCreatorWithKeyWriter returns a new DID document Creator. A Creator created with this function will automatically
+// generate keys for you when creating new DID documents. Those keys will be generated and stored using the given
+// KeyWriter. See the Create method for more information.
+func NewCreatorWithKeyWriter(keyWriter api.KeyWriter) (*Creator, error) {
+	if keyWriter == nil {
+		return nil, errors.New("a KeyWriter must be specified")
+	}
+
+	return &Creator{
+		keyWriter: keyWriter,
+	}, nil
 }
 
-// NewDIDCreator returns a new DIDCreator.
-// keyHandleReader is optional. If provided, then it will be used to fetch a key (based on the KeyID in the options)
-// in the Create method. If not provided, then a new key will be generated and used for DID creation.
-func NewDIDCreator(keyHandleReader api.KeyHandleReader) *DIDCreator {
-	return &DIDCreator{
-		keyHandleReader: keyHandleReader,
+// NewCreatorWithKeyReader returns a new DID document Creator. A Creator created with this function can be used to
+// create DID documents using your own already-generated keys from the given KeyReader.
+// At least one of keyHandleCreator and keyReader must be provided. See the Create method for more information.
+func NewCreatorWithKeyReader(keyReader api.KeyReader) (*Creator, error) {
+	if keyReader == nil {
+		return nil, errors.New("a KeyReader must be specified")
 	}
+
+	return &Creator{
+		keyReader: keyReader,
+	}, nil
 }
 
-// Create creates a DID document for the given DID method.
+// Create creates a DID document using the given DID method.
 // The usage of createDIDOpts depends on the DID method you're using.
-func (d *DIDCreator) Create(method string, createDIDOpts *api.CreateDIDOpts) (*did.DocResolution, error) {
-	if method == DIDMethodKey || method == fmt.Sprintf("%s%s", didMethodPrefix, DIDMethodKey) {
-		return d.createKeyDIDDoc(createDIDOpts)
+//
+// For creating did:key documents, there are two relevant options that can be set in the createDIDOpts object: KeyID and
+// VerificationType.
+//
+//	If the Creator was created using the NewCreatorWithKeyWriter function, then both of those options are ignored.
+//	An ED25519 key will be generated and saved automatically, and the verification type will automatically be set to
```

Comment:
same here, mention this is the default and add a TODO note to support more key types.

Reply:
Done in https://github.com/trustbloc/wallet-sdk/pull/52